README.md

```diff
@@ -2,91 +2,70 @@
 
-```console
-$ helm repo add deis https://github.com/deis/charts
-```
+To use it yourself, you must have [make](https://www.gnu.org/software/make/) installed and Docker installed and running on your local development machine.
 
-Then, install the Deis chart!
+If you don't have Docker installed, please go to https://www.docker.com/ to install it.
 
-```console
-$ helm install deis/deis
-```
+After you have those dependencies, build your code with `make build` and execute unit tests with `make test`.
 
-The chart will install the entire Deis platform onto Kubernetes. You can monitor all the pods that it installs by running:
+## Native Go Development Environment
 
-```console
-$ kubectl get pods --namespace=deis
-```
+You can also use the standard `go` toolchain to build and test if you prefer. To do so, you'll need [glide](https://github.com/Masterminds/glide) 0.9 or above and [Go 1.6](http://golang.org) or above installed.
 
-Once this is done, SSH into a Kubernetes minion, and run the following:
+After you have those dependencies, you can build and unit-test your code with `go build` and `go test $(glide nv)`, respectively.
 
-```
-$ curl -sSL http://deis.io/deis-cli/install.sh | sh
-$ sudo mv deis /bin
-$ kubectl get service deis-controller
-$ deis register 10.247.59.157 # or the appropriate CLUSTER_IP
-$ ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
-$ eval $(ssh-agent) && ssh-add ~/.ssh/id_rsa
-$ deis keys:add ~/.ssh/id_rsa.pub
-$ deis create --no-remote
-Creating Application... done, created madras-radiator
-$ deis pull deis/example-go -a madras-radiator
-Creating build... ..o
-```
+Note that you will not be able to build or push Docker images using this method of development.
 
-If you want to hack on a new feature, rebuild your code, build the deis/builder image and push it to a Docker registry. The `$DEIS_REGISTRY` environment variable must point to a registry accessible to your Kubernetes cluster. If you're using a locally hosted Docker registry, you may need to configure the Docker engines on your Kubernetes nodes to allow `--insecure-registry 192.168.0.0/16` (or the appropriate address range).
+# Testing
 
-```console
-$ make build docker-build docker-push
-```
+The Deis project requires that as much code as possible is unit tested, but the core contributors also recognize that some code must be tested at a higher level (functional or integration tests, for example).
 
-Next, you'll want to remove the `deis-builder` [replication controller](http://kubernetes.io/v1.1/docs/user-guide/replication-controller.html) and re-create it to run your new image.
+The [end-to-end tests](https://github.com/deis/workflow-e2e) repository has our integration tests. Additionally, the core contributors and members of the community also regularly dogfood the platfomr. Since this particular component is at the center of much of the Deis platform, we find it especially important to dogfood it.
```

Comment:
Fixed typo in 9aaeef2ce6541460a6f3a7d26bad919f7eca70f1

Also, good idea on linking. Fix coming